woocommerce/woocommerce-admin
# Increase chart selection to 10 and allow color overriding

Chart colors are currently limited to five colors, this makes it hard to use charts in custom reports.

To improve chart colors this PR increases the limit of max chart items from five to ten and adds additional colors.

To allow developers to use custom chart colors this PR introduces a new filter to allow overriding the existing color pallet.

### Screenshots

![Screen Shot 2022-02-04 at 13 05 20](https://user-images.githubusercontent.com/1199991/152527059-aeb3bf34-ea50-43c9-96cc-a3a2a4990283.png)
![Screen Shot 2022-02-04 at 13 05 05](https://user-images.githubusercontent.com/1199991/152527065-41600185-6896-475d-8ff7-07f72bac1dc0.png)


### Detailed test instructions:

#### Test new colors
-   Run `npm run storybook`
-  Open the [chart component story](http://192.168.0.16:6007/?path=/story/woocommerce-admin-components-chart--default)
- See the new colors

#### Test color filter
- Add the following JS to your admin head. You can use a plugin like "Add Admin Javascript" to do this:
```
 addFilter( 'woocommerce_admin_chart_item_color', 'example', ( index, key, orderedKeys ) => '#7f54b3' );
```
- Navigate to the profile wizard. `wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Fproducts`.
- Make sure the chart line colors are purple.
